core/services/blockhashstore/feeder.go

```diff
@@ -114,3 +132,114 @@ func (f *Feeder) Run(ctx context.Context) error {
 	f.lastRunBlock = latestBlock
 	return errs
 }
+
+func (f *Feeder) runTrusted(
+	ctx context.Context,
+	latestBlock uint64,
+	fromBlock uint64,
+	blockToRequests map[uint64]map[string]struct{},
+) error {
+	var errs error
+
+	// Iterate through each request block via waitGroup.
+	// For blocks with pending requests, add them to the batch to be stored.
+	// Note: Golang maps sort items in a range randomly, so although the batch size is used
+	// to limit blocks-per-batch, every block has an equal chance of getting picked up
+	// on each run.
+	var batch = make(map[uint64]struct{})
```

Comment:
Trying to grok the logic: on each run of `runTrusted` we end up storing a batch of blockhashes up to the configured batch size?

Alternatively, could we make this 

```golang
var batches [][]uint64
```

And end up processing all the `blockToRequests` in one iteration of this loop, potentially sending out multiple batches if needed? Or am I understanding this wrong?

Reply:
I guess we can send them all out in multiple transactions, could be quicker.